app/scripts/app.js

```diff
@@ -485,6 +485,118 @@ angular
       LabelFilter.readPersistedState();
     });
   })
+  .run(function($location, $uibModal, AuthService, Constants, countdownToTimestampFilter) {
+    if (Constants.INACTIVITY_TIMEOUT_PERIOD <= 0) {
+      return;
+    }
+    var lastInteractionKey = 'origin-web-console-last-interaction-timestamp';
+    var inactivityLogoutKey = 'origin-web-console-inactivity-logout';
+    var isLocalLogoutModalShown = false;
+    var checkInteractionInterval;
+    var modalInstance;
+    // Interval that will check for user inactivity every minute.
+    // In case the last interaction is higher then  INACTIVITY_TIMEOUT_PERIOD constant a logout modal will be shown.
+    // Also check for 'origin-web-console-inactivity-logout' object in the local storage, that indicates that the
+    // user has been already logged out, in case of multiple opened console tabs.
+    var startInteractionIntervalCheck = function () {
+      checkInteractionInterval = setInterval(function(){
+        if (!AuthService.isLoggedIn()) {
+          return;
+        }
+        var lastInteraction = Date.parse(localStorage.getItem(lastInteractionKey));
+        if (isNaN(lastInteraction)) {
+          Logger.warn("Last interaction timestamp has been corrupted. The logout timeout will be restarted.");
+          lastInteraction = new Date();
+        }
+        var currentTime = new Date();
+        // Since the INACTIVITY_TIMEOUT_PERIOD is set in minutes it needs to be converted into milliseconds.
+        if (currentTime - lastInteraction > Constants.INACTIVITY_TIMEOUT_PERIOD * 60000) {
+          showLogoutModal();
+        }
+      }, 60000);
+    };
+
+    // In case of multiple opened tabs the interval needs to be restarted everytime there is any activity in any tab,
+    // so all intervals are synced, and they start the counting at the same time.
+    var restartCheckInteractionInterval = function() {
+      // Handles cases of user activity during the opened modal, where user clicks inside the modal but not on the buttons.
+      if (modalInstance) {
+        modalInstance.dismiss("User activity");
+        modalInstance = null;
+      }
+      clearInterval(checkInteractionInterval);
+      startInteractionIntervalCheck();
+    };
+    // Updates lastInteraction date and restarts interval
+    var updateLastInteraction = function() {
+      restartCheckInteractionInterval();
```

Comment:
looks like another case where restartCheckInteractionInteral doesnt need to be called directly since it will get called from the event

Reply:
We need to call it here because the listener on the `storage` event is not fired in the tab where the event happened.
Removing `updateLastInteraction()` from the the modal resolving block remove the cases where the `restartCheckInteractionInterval()` is called twice (tested it)

Reply:
Ok good to know